File: course_scheduling/coursescheduling.cpp
#include <iostream>
#include <map>
#include <set>  // Unoreder set would be better


int main() {
    int n; std::cin >> n;
    std::map<std::string, std::set<std::string>> courses;
    for (int i=0; i<n; i++) {
        std::string firstName, lastName, course;
        std::cin >> firstName >> lastName >> course;
        std::string name = firstName.append(lastName);
        if (courses.find(course) == courses.end()) {
            std::set<std::string> name_set;
            courses[course] = name_set;
            courses[course].insert(name);
        } else {
            if (courses[course].find(name) == courses[course].end()) {
                courses[course].insert(name);
            }
        }
    }

    for (auto &course : courses) {
        std::cout << course.first << " " << course.second.size() << std::endl;
    }

    return 0;
}

File: course_scheduling/coursescheduling.py
# Note: Can use defaultdict to speed things up
#       Can do it using only 1 set (SM's answer)

def main():
    courses = dict()
    name_check = set()

    n = int(input())
    for i in range(n):
        first_name, last_name, course = input().split()
        name_course = first_name + last_name + course
        if name_course in name_check:
            continue
        else:
            name_check.add(name_course)
            if course not in courses.keys():
                courses[course] = 1
            else:
                courses[course] += 1

    course_counts = [(k, v) for k, v in courses.items()]
    course_counts.sort(key=lambda x: x[0])

    for entry in course_counts:
        print(f"{entry[0]} {entry[1]}")


if __name__ == "__main__":
    main()